File: Folicon_Native/Classes/PngToIcoService.vb

Imports System.Drawing
Imports System.Drawing.IconLib

Public Class PngToIcoService
    Public Shared Sub Convert(bitmap As Bitmap, ByVal icoPath As String)
        Dim mIcon As New MultiIcon()
        mIcon.Add("Untitled").CreateFrom(bitmap, IconOutputFormat.Vista)
        mIcon.SelectedIndex = 0
        mIcon.Save(icoPath, MultiIconFormat.ICO)
    End Sub
End Class

File: FoliCon/Modules/PngToIcoService.cs
namespace FoliCon.Modules;

public static class PngToIcoService
{
    public static void Convert(Bitmap bitmap, string icoPath)
    {
        var mIcon = new MultiIcon();
        mIcon.Add("Untitled").CreateFrom(bitmap, IconOutputFormat.Vista);
        mIcon.SelectedIndex = 0;
        mIcon.Save(icoPath, MultiIconFormat.ICO);
    }
}